controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/AbstractRequestProcessor.java

```diff
@@ -66,8 +67,15 @@
 @Slf4j
 public abstract class AbstractRequestProcessor<T extends ControllerEvent> extends SerializedRequestHandler<T> 
         implements StreamRequestProcessor {
-    protected static final Predicate<Throwable> OPERATION_NOT_ALLOWED_PREDICATE = e -> Exceptions.unwrap(e) 
-            instanceof StoreException.OperationNotAllowedException;
+    protected static final Predicate<Throwable> OPERATION_NOT_ALLOWED_PREDICATE = e -> Exceptions.unwrap(e) instanceof StoreException.OperationNotAllowedException;
+    protected static final Predicate<Throwable> ILLEGAL_STATE_PREDICATE = e -> Exceptions.unwrap(e) instanceof StoreException.IllegalStateException;
+    protected static final Predicate<Throwable> DATA_NOT_FOUND_PREDICATE = e -> Exceptions.unwrap(e) instanceof StoreException.DataNotFoundException;
+    protected static final Predicate<Throwable> VM_ERROR_PREDICATE = e -> Exceptions.unwrap(e) instanceof java.lang.VirtualMachineError;
+    protected static final Predicate<Throwable> SEGMENT_NOT_FOUND_PREDICATE = e -> Exceptions.unwrap(e) instanceof StoreException.DataContainerNotFoundException;
+    protected static final Predicate<Throwable> EVENT_RETRY_PREDICATE = OPERATION_NOT_ALLOWED_PREDICATE.or(ILLEGAL_STATE_PREDICATE).or(DATA_NOT_FOUND_PREDICATE)
```

Comment:
retry predicate should be:
OPERATION_NOT_ALLOWED_PREDICATE || not(ILLEGAL_STATE_PREDICATE) || not(DATA_NOT_FOUND_PREDICATE)
we dont want to retry on data_not_found predicate or illegal state predicate for instance.

Reply:
Perhaps you missed noticing that I have added a `negate()` at the end to indicate this predicate returns "false" if any of these conditions match.

Reply:
if its negate, then operationnotallowed should be removed from here.. we want to retry on operationnotallowed as that is thrown if the state transition is not allowed (like if its in scaling and a workflow is trying to set it to committing_txn).

Reply:
Ok. Removed OPERATION_NOT_FOUND from Predicate.